electron/src/renderer/App.jsx

```diff
@@ -0,0 +1,104 @@
+import { useState, useEffect, useContext } from 'react';
+import {
+  MemoryRouter as Router,
+  Switch,
+  Route,
+  Link,
+  Redirect,
+  useHistory,
+  useLocation,
+} from 'react-router-dom';
+import './App.css';
+import { realmApp, Authentication } from './realm';
+import Context from './components/Context';
+import LogIn from './components/LogIn';
+import SignUp from './components/SignUp';
+import Todo from './components/Todo';
+import Welcome from './components/Welcome';
+
+const PrivateRoute = (props) => {
+  const location = useLocation();
+  const { isLoggedIn } = useContext(Context);
+
+  return isLoggedIn ? (
+    <Route {...props} />
+  ) : (
+    <Redirect
+      to={{
+        pathname: '/',
+        state: { from: location },
+      }}
+    />
+  );
+};
+
+const App = () => {
+  const history = useHistory();
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+  useEffect(() => {
+    console.log(realmApp.currentUser?.isLoggedIn);
+    setIsLoggedIn(!!realmApp.currentUser?.isLoggedIn);
```

Comment:
personally not a big fan of using the double exclamation mark (!! or not not), especially here since the templates are intended to appeal to the broadest audience. A simpler alternative is something like:

`` app.currentUser?.isLoggedIn == true? setIsLoggedIn(true) : setIsLoggedIn(false)``

in the above example, if app.currentUser?.isLoggedIn == undefined, then the program will call setIsLoggedIn(false)

Reply:
> in the above example, if app.currentUser?.isLoggedIn == undefined, then the program will call setIsLoggedIn(false)
yes, and this is correct behavior 

anyways, can refactor to use ternary for readability